3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/CurrencyConversionEnrichment.scala

```
@@ -0,0 +1,155 @@
+          case Success("ENTERPRISE") => EnterpriseAccount.success
+          case Success("UNLIMITED")  => UnlimitedAccount.success
+
+          // These failures should be prevented by the schema
+          case Success(s) => "accountType [%s] is not one of DEVELOPER, ENTERPRISE, and UNLIMITED".format(s).toProcessingMessage.fail
+          case Failure(f) => Failure(f)
+        })
+        rateAt        <- ScalazJson4sUtils.extract[String](config, "parameters", "rateAt")
+        enrich        =  CurrencyConversionEnrichment(accountType, apiKey, baseCurrency, rateAt)
+      } yield enrich).toValidationNel
+    })
+  }
+}
+
+/**
+ * Configuration for a currency_conversion enrichment
+ *
+ * @param apiKey OER authentication
+ * @param baseCurrency Currency to which to convert
+ * @param rateAt Which exchange rate to use - "EOD_PRIOR" for "end of previous day".
+ */
+case class CurrencyConversionEnrichment(
+  accountType: AccountType,
+  apiKey: String,
+  baseCurrency: String,
+  rateAt: String) extends Enrichment {
+
+  val version = new DefaultArtifactVersion("0.1.0")
+
+  val fx = Forex(ForexConfig(nowishCacheSize = 0, nowishSecs = 0, eodCacheSize = 0), OerClientConfig(apiKey, accountType))
+
+  /**
+   * Attempt to convert if the initial currency and value are both defined
+   *
+   * @param inputCurrency Option boxing the initial currency if it is present
+   * @param value Option boxing the amount to convert
+   * @return None.success if the inputs were not both defined,
+   *         otherwise Validation[Option[_]] boxing the result of the conversion
+   */
+  private def performConversion(initialCurrency: Option[String], value: Option[Double], tstamp: DateTime): Validation[String, Option[String]] =
+    (initialCurrency, value) match {
+      case (Some(ic), Some(v)) => fx.convert(v, ic).to(baseCurrency).at(tstamp) match {
+        case Left(l) => (s"Open Exchange Rates error, message: ${l.errorMessage}").failure
```

Could we report the OER error type too? https://github.com/snowplow/scala-forex/blob/master/src/main/scala/com.snowplowanalytics/forex/oerclient/OerResponseError.scala